mod_form.php

```diff
@@ -100,39 +107,40 @@ public function definition() {
             }
         }
 
+        $inputstyle = array('style' => 'width: 399px');
+
         // Form start.
         $mform =& $this->_form;
         $mform->addElement('hidden', 'course', $id);
         $mform->addElement('header', 'general', get_string('general', 'form'));
-        $mform->addElement('static', 'overview', get_string('overview', 'accredible'),
-            get_string('activitydescription', 'accredible'));
-        if ($alreadyexists) {
-            $mform->addElement('static', 'additionalactivitiesone', '', get_string('additionalactivitiesone', 'accredible'));
-        }
-        $mform->addElement('text', 'name', get_string('activityname', 'accredible'), array('style' => 'width: 399px'));
+
+        $mform->addElement('text', 'name', get_string('activityname', 'accredible'), $inputstyle);
         $mform->addRule('name', null, 'required', null, 'client');
         $mform->setType('name', PARAM_TEXT);
         $mform->setDefault('name', $course->fullname);
 
         if ($alreadyexists) {
-            $mform->addElement('static', 'additionalactivitiestwo', '', get_string('additionalactivitiestwo', 'accredible'));
+            $mform->addElement('static', 'additionalactivitiesone', '', get_string('additionalactivitiesone', 'accredible'));
         }
 
-        // If we're updating and have a group then let the issuer choose to edit this.
+        // Load available groups.
+        $templates = array('' => '') + $groupsclient->get_groups();
+        $mform->addElement('select', 'groupid', get_string('accrediblegroup', 'accredible'), $templates, $inputstyle);
+        $mform->addRule('groupid', null, 'required', null, 'client');
```

Comment:
When a group is chosen from the dropdown, should we check the checkbox next to the users who have already taken the course?
Also, for a group that does not allow duplicates, the error should be handled. Right now it fails with exception.